core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -58,8 +58,7 @@
   // AppEngine runtimes have constraints on threading and socket handling
   // that need to be accommodated.
   public static final boolean IS_RESTRICTED_APPENGINE =
-      System.getProperty("com.google.appengine.runtime.environment") != null
-          && "1.7".equals(System.getProperty("java.specification.version"));
+      System.getProperty("com.google.appengine.runtime.environment") != null;
```

Comment:
The ciphers and TLS version are dictated by [the ConnectionSpec](https://github.com/grpc/grpc-java/blob/v1.4.0/okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java#L79) which is [applied elsewhere](https://github.com/grpc/grpc-java/blob/v1.4.0/okhttp/src/main/java/io/grpc/okhttp/OkHttpTlsUpgrader.java#L75). ConnectionSpec [explicitly sets both tls versions and the suites](https://github.com/grpc/grpc-java/blob/v1.4.0/okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/ConnectionSpec.java#L128).

So the provider here will be required to support TLSv1.2 and a GCM cipher.